src/autora/doc/pipelines/train.py

```diff
@@ -0,0 +1,88 @@
+from typing import Dict, Iterable
+
+import torch
+from datasets import Dataset
+from peft import LoraConfig
+from transformers import AutoModelForCausalLM, AutoTokenizer, TrainingArguments
+from trl import SFTTrainer
+
+from autora.doc.pipelines.data import load_data, preprocess_code
+from autora.doc.runtime.predict_hf import get_quantization_config
+from autora.doc.runtime.prompts import INSTR_SWEETP_1, SYS_GUIDES, PromptBuilder
+
+
+def get_dataset(jsonl_path: str) -> Dataset:
+    # "instruction", "output"
+    inputs, labels = load_data(jsonl_path)
+
+    def gen() -> Iterable[Dict[str, str]]:
+        for i, o in zip(inputs, labels):
+            text = PromptBuilder(SYS_GUIDES, INSTR_SWEETP_1).add_example(preprocess_code(i), o).build()
+            yield {"text": text}
+
+    ds = Dataset.from_generator(gen)
+    return ds
+
+
+def fine_tune(base_model: str, new_model_name: str, dataset: Dataset) -> None:
+    cuda_available = torch.cuda.is_available()
+    # train using 4 bit quantization for lower GPU memory usage
+    kwargs = (
+        {"device_map": "auto", "quantization_config": get_quantization_config()} if cuda_available else {}
+    )
```

Comment:
```suggestion
    kwargs = {}
    if cuda_available:
        kwargs.update({"device_map": "auto", "quantization_config": get_quantization_config()})
```

Small update to the code for more readability since the default of empty kwargs is not very visible.